mathesar/pagination.py

```diff
@@ -39,5 +40,42 @@ def paginate_queryset(self, queryset, request, table_id, filters=[], order_by=[]
         table = queryset.get(id=table_id)
         self.count = table.sa_num_records
         self.request = request
-        return table.get_records(self.limit, self.offset,
-                                 filters=filters, order_by=order_by)
+
+        return table.get_records(
+            self.limit, self.offset, filters=filters, order_by=order_by,
+        )
+
+
+class TableLimitOffsetGroupPagination(TableLimitOffsetPagination):
+    def get_paginated_response(self, data):
+        return Response(OrderedDict([
+            ('count', self.count),
+            ('group_count', self.group_count),
+            ('results', data)
+        ]))
+
+    def paginate_queryset(self, queryset, request, table_id,
+                          filters=[], order_by=[], group_count_by=[]):
+        records = super().paginate_queryset(
+            queryset, request, table_id, filters=filters, order_by=order_by
+        )
+
+        table = queryset.get(id=table_id)
+        if group_count_by:
+            group_count = table.get_group_counts(
+                group_count_by, self.limit, self.offset,
+                filters=filters, order_by=order_by
+            )
+            # Convert the tuple keys into strings so it can be converted to JSON
+            group_count = {','.join(k): v for k, v in group_count.items()}
```

Comment:
This is a bug if a string rather than a tuple is returned by `table.get_group_counts`, i.e., if we want to group by only one column.  I suggest modifying that function so it only has one return type(ish), a tuple.  This will be easier to deal with, and less prone to causing bugs.  There should also be a test for that.

Reply:
In particular, this results in splitting the value of that column up into its constituent characters in the output.

Reply:
![image](https://user-images.githubusercontent.com/6829750/124590893-3cb2b880-de8e-11eb-87a8-3c4212d745d7.png)

Reply:
Good catch, thanks